airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/PersistStateActivityImpl.java

```diff
@@ -39,4 +48,17 @@ public boolean persist(final UUID connectionId, final StandardSyncOutput syncOut
     }
   }
 
+  private static void validateStreamStates(final StateWrapper state, final ConfiguredAirbyteCatalog configuredCatalog) {
+    final List<StreamDescriptor> stateStreamDescriptors =
+        state.getStateMessages().stream().map(stateMessage -> stateMessage.getStream().getStreamDescriptor()).toList();
+    configuredCatalog.getStreams().forEach(stream -> {
+      final StreamDescriptor streamDescriptor = new StreamDescriptor().withName(stream.getStream().getName())
+          .withNamespace(stream.getStream().getNamespace());
```

Comment:
nit: replace this with a call to `CatalogHelpers.extractDescriptor(stream)`

Reply:
@alovew see the above nit to use the CatalogHelpers method here

Reply:
oops missed this, just updated